internal/auth0/client/resource_test.go

```diff
@@ -2315,3 +2315,115 @@ func TestAccClientCanSetDefaultAuthMethodOnCreate(t *testing.T) {
+	depends_on = [ resource.auth0_organization.my_new_org ]
+	name = "temp-new-org"
+}
+
+resource "auth0_client" "my_client" {
+	depends_on = [ data.auth0_organization.my_new_org ]
+    name = "Acceptance Test - DefaultOrganization - {{.testName}}"
+    default_organization {
+        flows = ["client_credentials"]
+        organization_id = data.auth0_organization.my_new_org.id
+    }
+}
+`
+
+const testAccUpdateClientRemoveDefaultOrganization = `
+resource "auth0_client" "my_client" {
+    name = "Acceptance Test - DefaultOrganization - {{.testName}}"
+    default_organization {
+		disable = true
+    }
+}
+`
+
+// Const testAccUpdateClientDefaultOrganizationFlowsOnly = `
+//resource "auth0_client" "my_client" {
+//    name = "Acceptance Test - DefaultOrganization - {{.testName}}"
+//    default_organization {
+//		flows = ["client_credentials"]
+//    }
+//}
+//`
+//
+//const testAccUpdateClientDefaultOrganizationOrgIdOnly = `
+//resource "auth0_client" "my_client" {
+//    name = "Acceptance Test - DefaultOrganization - {{.testName}}"
+//    default_organization {
+//        organization_id = "org_z5YvxlXPO0NspoIa"
+//    }
+//}
+//`.
+
+func TestAccClientWithDefaultOrganization(t *testing.T) {
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ParseTestName(testAccCreateClientWithDefaultOrganization, t.Name()),
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckResourceAttr("auth0_client.my_client", "name", fmt.Sprintf("Acceptance Test - DefaultOrganization - %s", t.Name())),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "default_organization.0.flows.0", "client_credentials"),
+					resource.TestCheckResourceAttrSet("auth0_client.my_client", "default_organization.0.organization_id"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccUpdateClientWithDefaultOrganization, t.Name()),
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckResourceAttr("auth0_client.my_client", "name", fmt.Sprintf("Acceptance Test - DefaultOrganization - %s", t.Name())),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "default_organization.0.flows.0", "client_credentials"),
+					resource.TestCheckResourceAttrSet("auth0_client.my_client", "default_organization.0.organization_id"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccUpdateClientRemoveDefaultOrganization, t.Name()),
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckResourceAttr("auth0_client.my_client", "name", fmt.Sprintf("Acceptance Test - DefaultOrganization - %s", t.Name())),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "default_organization.0.disable", "true"),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "default_organization.0.flows.#", "0"),
+					resource.TestCheckResourceAttr("auth0_client.my_client", "default_organization.0.organization_id", ""),
+				),
+			},
+			//{
```

Comment:
@acwest For some reason, I'm not able to test this case. Let's have a quick word when you get a chance.